src/game.cpp

```diff
@@ -4102,6 +4102,22 @@ bool Game::combatChangeHealth(Creature* attacker, Creature* target, CombatDamage
 		target->gainHealth(attacker, damage.primary.value);
 		realHealthChange = target->getHealth() - realHealthChange;
 
+		if (attackerPlayer && attackerPlayer != targetPlayer) {
+			attackerPlayer->sendCombatAnalyzer(damage.primary.type, damage.primary.value, DamageAnalyzerImpactType::DEALT,
+				target ? target->getName() : "(other)");
+			if (damage.secondary.type == COMBAT_HEALING) {
+				attackerPlayer->sendCombatAnalyzer(damage.secondary.type, damage.secondary.value, DamageAnalyzerImpactType::DEALT,
+					target ? target->getName() : "(other)");
+			}
+		} else if (targetPlayer) {
+			targetPlayer->sendCombatAnalyzer(damage.primary.type, damage.primary.value, DamageAnalyzerImpactType::HEALING,
+				attacker ? attacker->getName() : "(other)");
+			if (damage.secondary.type == COMBAT_HEALING) {
```

Comment:
Isnt secondary damage only meant for elemental damage?

Reply:
> 

Yes, secondary damage can be [COMBAT_HEALING](https://github.com/otland/forgottenserver/blob/d19d63b556ebccc8b58f100849938223dbe47d86/src/enums.h#L180-L197)

Reply:
> 

Yes, secondary damage can be [COMBAT_HEALING](https://github.com/otland/forgottenserver/blob/d19d63b556ebccc8b58f100849938223dbe47d86/src/enums.h#L180-L197)